src/olympia/reviews/serializers.py

```diff
@@ -66,12 +70,13 @@ def validate(self, data):
             raise serializers.ValidationError(
                 'An add-on author can not leave a review on its own add-on.')
 
-        if Review.objects.filter(
-                addon=data['addon'], user=data['user'],
-                version=data['version']).exists():
-            raise serializers.ValidationError(
-                'The same user can not leave a review on the same version more'
-                ' than once.')
+        if not self.partial:
+            if Review.objects.filter(
+                    addon=data['addon'], user=data['user'],
+                    version=data['version']).exists():
```

Comment:
I really hate if-statements like that, can we do something like ``review_exists = Review.objects...`` and ``if review_exists: ...``?